mongomock/filtering.py

```diff
@@ -154,6 +154,13 @@ def _regex(doc_val, regex):
     return any(regex.search(item) for item in _force_list(doc_val))
 
 
+def _size_op(doc_val, search_val):
+    if isinstance(doc_val, (list, tuple, dict)):
+        return search_val == len(doc_val)
+    else:
+        return search_val == 1 if doc_val else 0
```

Comment:
i am not really comfortable in treating 1 as True, 0 as False. So   `s/return int(doc_val is not None)/return  doc_val is not None` should work.